pkg/registry/pod/etcd/etcd_test.go

```diff
@@ -130,6 +133,54 @@ func TestDelete(t *testing.T) {
 	test.TestDeleteGraceful(scheduledPod, 30)
 }
 
+type FailDeletionStorage struct {
+	storage.Interface
+}
+
+func (f FailDeletionStorage) Delete(ctx context.Context, key string, out runtime.Object) error {
+	return etcd.Error{Code: etcd.ErrorCodeKeyNotFound}
+}
+
+func newFailDeleteStorage(t *testing.T) (*REST, *etcdtesting.EtcdTestServer) {
+	etcdStorage, server := registrytest.NewEtcdStorage(t, "")
+	failDeleteStorage := FailDeletionStorage{etcdStorage}
+	restOptions := generic.RESTOptions{failDeleteStorage, generic.UndecoratedStorage, 3}
+	storage := NewStorage(restOptions, nil, nil)
+	return storage.Pod, server
+}
+
+func TestIgnoreDeleteNotFound(t *testing.T) {
+	pod := validNewPod()
+	testContext := api.WithNamespace(api.NewContext(), api.NamespaceDefault)
+	registry, server := newFailDeleteStorage(t)
+	defer server.Terminate(t)
+
+	fakeStorage := FailDeletionStorage{registry.Storage}
+	registry.Storage = fakeStorage
+	defer server.Terminate(t)
+
+	// should fail if pod A is not created yet.
+	_, err := registry.Delete(testContext, pod.Name, nil)
+	if !errors.IsNotFound(err) {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	// create pod
+	_, err = registry.Create(testContext, pod)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	// delete object with grace period 0, storage will return NotFound, but the
+	// registry shouldn't get any error since we ignore the NotFound error.
+	zero := int64(0)
+	opt := &api.DeleteOptions{GracePeriodSeconds: &zero}
+	_, err = registry.Delete(testContext, pod.Name, opt)
```

Comment:
And add a flag in the test storage to make sure our delete override got called and actually returned the error